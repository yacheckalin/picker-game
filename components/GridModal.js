import React, { useEffect } from "react";
import PropTypes from "prop-types";

const GridModal = ({ message, win = false }) => {
  useEffect(() => {
    const callback = () => {
      const elem = document.querySelector("#win-modal");
      const instance = M.Modal.init(elem, {});
    };
    document.addEventListener("DOMContentLoaded", callback);
    if (win) {
      const elem = document.querySelector("#win-modal");
      const instance = M.Modal.init(elem, {});
      instance.open();
    }
    return () => {
      document.removeEventListener("DOMContentLoaded", callback);
    };
  });
  return (
    <div id="win-modal" className="modal">
      <div className="modal-content center">
        <h4>{message}</h4>
      </div>
      <div className="modal-footer center">
        <a href="#!" className="modal-close waves-effect waves-green btn-flat">
          Close
        </a>
      </div>
    </div>
  );
};
GridModal.propTypes = {
  message: PropTypes.string,
  win: PropTypes.bool.isRequired,
};
GridModal.defaultProps = {
  win: false,
};

export default GridModal;
