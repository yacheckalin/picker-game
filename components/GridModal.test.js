import React from "react";
import { shallow, mount, render } from "enzyme";
import toJSON from "enzyme-to-json";

import GridModal from "./GridModal";
// import M from "materialize-css/dist/js/materialize";

describe("<GridModal />", () => {
  const props = {
    message: "Winning message!",
    win: false,
  };

  test("renders and displays properly", () => {
    const wrapper = shallow(<GridModal {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("render open modal", () => {
    const wrapper = render(<GridModal {...props} win={true} />);

    expect(wrapper.find(".modal-content").first().text()).toContain(
      props.message
    );
  });
});
