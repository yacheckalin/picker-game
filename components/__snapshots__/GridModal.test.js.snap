// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridModal /> renders and displays properly 1`] = `
<div
  className="modal"
  id="win-modal"
>
  <div
    className="modal-content center"
  >
    <h4>
      Winning message!
    </h4>
  </div>
  <div
    className="modal-footer center"
  >
    <a
      className="modal-close waves-effect waves-green btn-flat"
      href="#!"
    >
      Close
    </a>
  </div>
</div>
`;
